charts/valkey/templates/service.yaml

```diff
@@ -0,0 +1,48 @@
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ include "valkey.fullname" . }}
+  labels:
+    {{- include "valkey.labels" . | nindent 4 }}
+spec:
+  ports:
+    - name: client
+      port: {{ .Values.ports.client.port }}
+      targetPort: {{ .Values.ports.client.port }}
+    - name: cluster
+      port: {{ .Values.ports.cluster.port }}
+      targetPort: {{ .Values.ports.cluster.port }}
+    {{- if .Values.prometheusExporter.enable }}
+    - name: metrics
+      port: {{ .Values.prometheusExporter.port }}
+      targetPort: {{ .Values.prometheusExporter.port }}
+    {{- end}}
+  type: ClusterIP
+  clusterIP: None
+  selector:
+    {{- include "valkey.selectorLabels" . | nindent 4 }}
```

Comment:
Seems like if you only use a single service (why are there two?) and leave the `type` field empty, you can even upgrade from existing installations.

Here is the error if you apply this chart on an existing installation:

```
Error: UPGRADE FAILED: cannot patch "valkey-primary" with kind Service: Service "valkey-primary" is invalid: spec.clusterIPs[0]: Invalid value: ["None"]: may not change once set
```

When you set a name override to `valkey-primary` and remove the second service, the deployment passes.

Reply:
That's fine.